azure/scope/cluster.go

```diff
@@ -755,6 +755,19 @@ func (s *ClusterScope) ClusterName() string {
 	return s.Cluster.Name
 }
 
+// AzureClusterIdentityName returns the azure cluster identity name.
+func (s *ClusterScope) AzureClusterIdentityName() string {
+	if s.AzureCluster.Spec.IdentityRef == nil {
+		return ""
+	}
+	return s.AzureCluster.Spec.IdentityRef.Name
+}
+
+// SubscriptionID returns the Azure client Subscription ID.
+func (s *ClusterScope) SubscriptionID() string {
+	return s.AzureClients.SubscriptionID()
```

Comment:
Since AzureClients is embedded in ClusterScope, I don't think this method definition is strictly required to fulfill the GroupScope interface. Unless there's another need for this I'm not seeing.

Reply:
I've added it initially since the `managedcontrolplane` has it too. Removed it for now.